addons/base_import/models/base_import.py

```diff
@@ -453,10 +473,19 @@ def _match_headers(self, rows, fields, options):
             return [], {}
 
         headers = next(rows)
-        return headers, {
-            index: [field['name'] for field in self._match_header(header, fields, options)] or None
-            for index, header in enumerate(headers)
-        }
+        matches = {}
+        mapping_fields = self.env['base_import.mapping'].search_read([('res_model', '=', self.res_model)], ['column_name', 'field_name'])
+        for index, header in enumerate(headers):
+            match_fields = []
+            for mapping_field in mapping_fields:
+                if mapping_field['column_name'] == header.lower():
```

Comment:
mapping_fields should be a mapping of `{column_name: field_name}` so there is no need to repeatedly iterate over it.

The variables should also be renamed & the model should be documented as a mapping of previous header:field selections or some such.